zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -86,10 +86,16 @@
 public class ZooKeeperServer implements SessionExpirer, ServerStats.Provider {
     protected static final Logger LOG;
 
+    public static final String GLOBAL_OUTSTANDING_LIMIT = "zookeeper.globalOutstandingLimit";
+    protected static int globalOutstandingLimit = 1000;
```

Comment:
Findbug is complaining about this (none final static variable + overriding it in subclass). 

Two options: 
* We revert to the old way of handling `globalOutstandingLimit ` by accessing it indirectly through `getGlobalOutstandingLimit`. Thus we can override this method in sub class (we can't override a static variable in sub class).

* Mute Findbug for this issue in this class and its subclass. Though, it seems Findbug's concern in this case is legitimate so maybe first option is better?

@lvfangmin Would you like a following up pull request to address the find bug issues? It currently blocks new pull request builds from greening.

Reply:
@lvfangmin Please don't mute Findbugs. I think the original implementation with the overridable getter was a lot neater. Sorry, I didn't have a chance to review this patch and catch it in time.